File: server/src/main/java/com/school_system/service/ClientInfoService.java
package com.school_system.service;

import com.school_system.entity.school.ClientInfo;
import com.school_system.common.ResponseObject;
import com.school_system.entity.school.ClientPreferences;

public interface ClientInfoService {
    ResponseObject<ClientInfo> getClientInfo();
    ResponseObject<ClientInfo> createClientInfo(ClientInfo clientInfo);
    ResponseObject<ClientInfo> updateClientInfo(ClientInfo clientInfoRequest);
    ResponseObject<ClientPreferences> getPreferences();
}

File: webapp/src/app/_services/data/clientInfo.service.ts
import { Injectable } from '@angular/core';
import {ResponseObject} from "../../types/ResponseObject";
import {HttpClientService} from "../http-client-service";

import {ClientInfo} from "../../types/ClientInfo";


@Injectable({
	providedIn: "root",
})
export class ClientInfoService {

	

	constructor(private http: HttpClientService) {
	}
	
	getClientInfo() {
		return this.http.get<ResponseObject<ClientInfo>>("/clientInfo", true);
	}
	
	createClientInfo(data: any) {
		return this.http.post<ResponseObject<ClientInfo>>("/clientInfo", data);
	}
	
	updateClientInfo(data: any) {
		return this.http.put<ResponseObject<ClientInfo>>("/clientInfo", data);
	}
	
	
	getPreferences() {
		return this.http.get<ResponseObject<ClientInfo>>("/clientInfo/preferences");
		
		
	}
}
